apps/data-pipeline/ap-exams-importer/src/index.ts

```diff
@@ -50,12 +55,29 @@ async function main() {
           .then((rows) => rows[0]);
 
         for (const score of reward.acceptableScores) {
-          await tx.insert(apExamToReward).values({ examId: fullName, score, reward: rewardId });
+          await tx
+            .insert(apExamToReward)
+            .values({ examId: fullName, score, reward: rewardId })
+            .onConflictDoUpdate({
+              target: [apExamToReward.examId, apExamToReward.score],
+              set: conflictUpdateSetAllCols(apExamToReward),
+            });
         }
       }
     }
   });
-
+  //remove apExamRewards that don't exist in the junction table(ap_exam_to_reward)
+  //done using correlated subquery
+  await db
+    .delete(apExamReward)
+    .where(
+      notExists(
```

Comment:
You should investigate whether something of the form `DELETE FROM a WHERE a.pkey IN (SELECT a.pkey FROM a LEFT JOIN b ON ... WHERE b.pkey IS NULL)` is faster, for table `a` having a one-to-many relationship with `b`. I'm not sure myself, but I suspect the join-based proposal is faster. My intuition is that many `EXISTS`/`NOT EXISTS` on one row at a time is not as likely to optimize.